capa/features/extractors/viv/insn.py

```diff
@@ -109,6 +111,35 @@ def extract_insn_api_features(fh: FunctionHandle, bb, ih: InsnHandle) -> Iterato
         if not target:
             return
 
+        if f.vw.metadata["Format"] == "elf":
+            if "symtab" not in fh.ctx["cache"]:
+                # the symbol table gets stored as a function's attribute in order to avoid running
+                # this code everytime the call is made, thus preventing the computational overhead.
+                elf = f.vw.parsedbin
+                endian = "<" if elf.getEndian() == 0 else ">"
+                bitness = elf.bits
+
+                SHT_SYMTAB = 0x2
+                for section in elf.sections:
+                    if section.sh_info & SHT_SYMTAB:
+                        strtab_section = elf.sections[section.vsGetField("sh_link")]
+                        sh_symtab = Shdr.from_viv(section, elf.readAtOffset(section.sh_offset, section.sh_size))
+                        sh_strtab = Shdr.from_viv(strtab_section, elf.readAtOffset(strtab_section.sh_offset, strtab_section.sh_size))
+
+                symtab = SymTab(endian, bitness, sh_symtab, sh_strtab)
+                fh.ctx["cache"]["symtab"] = symtab
+
+            symtab = fh.ctx["cache"]["symtab"]
+            for symbol in symtab.get_symbols():
+                sym_name = symtab.get_name(symbol)
+                sym_value = symbol.value
+                sym_info = symbol.info
+
+                STT_FUNC = 0x2
+                if sym_value == target and sym_info & STT_FUNC != 0:
+                    yield API(sym_name), ih.address
+                    yield FunctionName(sym_name), target
```

Comment:
this should be extracted at function scope, for the target function. extracting it here sorta works, but its not correct.

imagine the case where we might ask capa: "show me all features for the function at 0x401000" and it extracts features only for that function. we'd want the `FunctionName` to be extracted for that function in that function, not when we encounter a call to that function.

so, please move this extraction into `viv/function.py`. this might also require parsing the symtab there, so it might make sense to refactor the symtab cache creation into a function that can be reused in both places.